docs/protocol.md

```diff
@@ -0,0 +1,333 @@
+  | ------------ | ------------ | --------------- |
+  | IPv6 address | 16           | Big-endian?     |
+  | Port         | 2            | Little-endian?! |
+
+The number of pairs in the message can vary, from 1 up to 8 peers.
+
+## Publish message
+
+Reference: class `rai::publish` in `rai/node/common.cpp`
+
+A Publish message simply contains a published Block, which was either created or
+forwarded by another node.
+
+See below for the way a block in encoded.
+
+## ConfirmReq message
+
+Reference: class `rai::confirm_req` in `rai/node/common.cpp`
+
+XXX A ConfirmReq message contains a Block for which a vote has been requested?
+
+  | Field | Size (bytes) | Description        |
+  | ----- | ------------ | ------------------ |
+  | Block | Varies       | A serialized Block |
+
+## ConfirmAck message
+
+Reference: class `rai::confirm_ack` in `rai/node/common.cpp`
+
+A ConfirmAck message contains a Vote by a representative on a Block.
+
+  | Field | Size (bytes) | Description       |
+  | ----- | ------------ | ----------------- |    
+  | Vote  | (varies)     | A serialized Vote |
+
+
+## Vote
+
+Reference: class `rai::vote` in `rai/common.cpp`
+
+  | Field     | Size (bytes) | Description                          |
+  | --------- | ------------ | ------------------------------------ |
+  | Account   | 32           | Public key of the voting Account     |
+  | Signature | 64           | Signature                            |
+  | Sequence  | 8            | 64-bit sequence number (big-endian?) |
+  | Block     | Varies       | The Block being voted on             |
+
+The Signature of a Vote signs the following list of hashables: `hash(Block)`, `Sequence`
+
+XXX what is the exact function of the sequence number?
+
+# Blocks
+
+Reference: `rai/lib/blocks.cpp`
+
+Blocks are encoded into a byte stream, with each field directly following
+the previous value. All blocks contain a Signature and a Work value, which follow
+the block-specific fields:
+
+  | Field     | Size (bytes) | Description                                |
+  | --------- | ------------ | ------------------------------------------ |
+  | *Block-specific fields* | |                                           |
+  | Signature | 64           | Ed25519 signature of the block's hashables |
+  | Work      | 8            | Unsigned 64-bit integer (little endian?)   |
+
+The "hashables" of a block are the set of field values that are fed into
+the Blake2b hashing algorithm in order to get a hash value for the block.
+This hash value is then signed with the public key of the account that generated the block
+and stored in the Signature field.
+
+XXX Work value
+
+The different block types are denoted by a Type value (see `enum class block_type` in `rai/lib/blocks.hpp`):
+
+  | Block type | Value |
+  | ---------- | ----- |
+  | Send       | 2     |
+  | Receive    | 3     |
+  | Open       | 4     |
+  | Change     | 5     |
+  | State      | 6     |
```

Comment:
Maybe some restructuring here is a good idea. To emphasize that state blocks are to be used in all situations and other types are obsolete... Perhaps, "state blocks" and "legacy block types" subsections or something?